src/modules/CurrencyModule.cpp

```diff
@@ -1198,7 +1202,55 @@ CurrencyModule::CurrencyModule(UmikoBot* client) : Module("currency", true), m_c
 		auto& victimCurrency = guildList[channel.guildId()][getUserIndex(channel.guildId(), victimId)];
 		auto& authorCurrency = guildList[channel.guildId()][getUserIndex(channel.guildId(), authorId)];
 
-		if (roll)
+		EventModule* eventModule = static_cast<EventModule*>(UmikoBot::Instance().GetModuleByName("event"));
+		auto& eventConfig = eventModule->getEventServerData(channel.guildId());
+
+		if (roll && eventConfig.eventHighRiskHighRewardRunning)
+		{
+			QString num = QString::number(highRiskRewardBonus);
+			victimCurrency.setCurrency(victimCurrency.currency() - amountToSteal);
+			authorCurrency.setCurrency(authorCurrency.currency() + amountToSteal + highRiskRewardBonus);
+
+			QString stealAmount = QString::number(amountToSteal);
+			QString output = QString(
+				":man_detective: **Steal success!** :man_detective:\n"
+				"*%1* has discreetly stolen **`%2 %3`** from under *%4's* nose.\n"
+				"HighRiskHighReward event **BONUS**: `%5 %3`\n"
+			).arg(thiefName, stealAmount, config.currencySymbol, victimName, num);
+
+			client.createMessage(message.channelId(), output);
+			return;
+		}
+
+		if (amountToSteal < lowRiskRewardLeastAmoutToSteal && eventConfig.eventLowRiskLowRewardRunning)
+		{
+			QString num = QString::number(lowRiskRewardLeastAmoutToSteal);
+			QString desc = QString(
+				"**You can't steal that amount beacuse LowRiskLowReward event is running!**\n"
+				"*(You have to steal more than or equal to `%2 %1`)*\n"
```

Comment:
Because the LowRiskLowReward event issues a penalty if the user succeeds in stealing! The least amount is there to stop losing money by stealing, suppose you steal 10 CC from a person, you succeeded, now in LowRiskLowReward event you will be charged 50 CC (approximately) for succeeding so you will lose 40 CC if you try to steal 10 CC and succeed! To prevent this the least steal amount is introduced in LowRiskLowReward event! 🙂

Reply:
Shouldn't the penalty be proportional to the amount stolen?

Reply:
Nice idea, I will try to implement that!

Reply:
Done!